src/svg/qsvggenerator.cpp

```diff
@@ -998,7 +1017,36 @@ void QSvgPaintEngine::updateState(const QPaintEngineState &state)
     if (d->afterFirstUpdate)
         *d->stream << "</g>\n\n";
 
-    *d->stream << "<g ";
+	QPainter* p = painter();
```

Comment:
If you test the `flags` variable before they update it with `flags |= QPaintEngine::AllDirty;` above, then you can check the flags for `QPaintEngine::DirtyClipPath` and extract the clip path directly from `state->clipPath()` (https://doc.qt.io/qt-5/qpaintenginestate.html#clipPath). This would prevent writing a new clip path every time the state is updated, when it may not in fact have changed. You would still need to write the most recent path id into the <g> elements each time, but at least then they would be reusing it.